packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -340,30 +340,20 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection impleme
     /**
      * Disconnect from the websocket
      */
-    public disconnect(socketProtocolError: boolean = false) {
-        // We set the closed flag as a part of the contract for overriding the disconnect method.  This is used by
-        // DocumentDeltaConnection to determine if emitting on the socket is allowed, which is important since
-        // OdspDocumentDeltaConnection reuses the socket rather than truly disconnecting it.  Note that below we may
-        // still send disconnect_document which is allowed; this is only intended to prevent normal messages from
-        // being emitted.
-        this.closed = true;
-
-        if (this.socketReferenceKey !== undefined) {
-            const key = this.socketReferenceKey;
-            this.socketReferenceKey = undefined;
-
-            const reason = "client closing connection";
-
-            if (this.enableMultiplexing && !socketProtocolError && this.hasDetails) {
-                // tell the server we are disconnecting this client from the document
-                this.socket.emit("disconnect_document", this.clientId, this.documentId);
-            }
+    protected disconnect(socketProtocolError: boolean, reason: DriverError) {
+        const key = this.socketReferenceKey;
+        assert(key !== undefined, "reetrancy not supported!");
+        this.socketReferenceKey = undefined;
+
+        if (!socketProtocolError && this.hasDetails) {
+            // tell the server we are disconnecting this client from the document
+            this.socket.emit("disconnect_document", this.clientId, this.documentId);
+        }
 
-            OdspDocumentDeltaConnection.removeSocketIoReference(key, socketProtocolError, reason);
+        OdspDocumentDeltaConnection.removeSocketIoReference(key, socketProtocolError, reason);
 
-            // RemoveSocketIoReference() above raises "disconnect" event on socket for socketProtocolError === true
-            // If it's not critical error, we want to raise event on this object only.
-            this.emit("disconnect", reason);
-        }
+        // RemoveSocketIoReference() above raises "disconnect" event on socket for socketProtocolError === true
+        // If it's not critical error, we want to raise event on this object only.
+        this.emit("disconnect", reason);
```

Comment:
I think currently this causes the DeltaConnection to close itself, but the handler is removed now.  Is this behavior important to preserve?

Reply:
DeltaConnection is benign pass-through (in this PR), it does not do anything.
Also handlers are removed on socket (prior to getting here), not on DocumentDeltaConnection instance (or derived class) - this is now (after your feedback) moved to be responsibility of the caller/user of these classes.

Please let me know if I got it wrong.